Virtualization causes TaskCanceledExceptions after a while

**Describe the bug**
the problem is exactly like described in https://github.com/dotnet/aspnetcore/issues/30086

Its marked "Closed" there but it still happens now.
Just wondering if its a "Virtualization" problem or something with the Datagrid.


```
28.12.2021 16:08:53 [WARN] [Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer] - Unhandled exception rendering component: A task was canceled. 
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
   at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
   at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
28.12.2021 16:08:53 [ERROR] [Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost] - Unhandled exception in circuit 'A0-mrmrbUWgkxne8pD_evDxBQhUd6fXZ-4sAwj_jjEY'. 
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
   at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
   at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
```

**To Reproduce**
1. Use the example below.
2. Open the modal and scroll around a bit
3. Close the page
4. Wait about a minute or two.
5. Exception above should log.


**Expected behavior**
no exception

**Example**
```
<Modal @ref="myModal">
    <ModalContent Centered="true" Size="ModalSize.ExtraLarge">
        <ModalHeader>
            <ModalTitle>
                Some Modal Title
            </ModalTitle>
        </ModalHeader>
        <ModalBody>
            <Row>
                <Column ColumnSize="ColumnSize.Is3">
                    <Card>
                        <CardHeader>
                            <CardTitle>Some Card Title</CardTitle>
                        </CardHeader>
                        <CardBody Padding="Padding.Is0" Overflow="Overflow.Auto">
                            <DataGrid TItem="SomeType" Data="@dataList" Narrow="true" FixedHeader="true" Virtualize="true"
                                      VirtualizeOptions="@(new VirtualizeOptions() { DataGridHeight="calc(-8px + 70vh) !important", DataGridMaxHeight="calc(-8px + 70vh) !important" })"
                                      Striped="true" Bordered="true" Filterable="true" Sortable="false" SelectionMode="DataGridSelectionMode.Multiple"
                                      Responsive="true"
                                      SelectedRow="null"
                                      SelectedRows="@myList"
                                      SelectedRowsChanged="@AddOrRemoveMyList">
                                <DataGridColumn TItem="SomeType" Field="@nameof(SomeType.MyProperty1)" Caption="No" Sortable="false" Filterable="false" />
                                <DataGridColumn TItem="SomeType" Field="@nameof(SomeType.Propberty2)" Caption="Name" Sortable="false" Filterable="false" />
                            </DataGrid>
                        </CardBody>
                    </Card>
                </Column>
            </Row>
        </ModalBody>
        <ModalFooter>
            <Button Color="Color.Danger" Clicked="@ResetSelection">Reset</Button>
        </ModalFooter>
    </ModalContent>
</Modal>

<Button Color="Color.Danger" Clicked="@OpenModal">Show Modal</Button>

@code {
    private List<SomeType> dataList = new List<SomeType>() { 
        new SomeType(1, "a"),
        new SomeType(2, "b"),
        new SomeType(3, "c"),
        new SomeType(4, "d"),
        new SomeType(5, "e"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
        new SomeType(6, "f"),
    };
    private List<SomeType> myList = new List<SomeType>();
    private Modal myModal;

    private async Task OpenModal()
    {
        await myModal.Show();
        myList = dataList.Where(x => x.MyProperty1 == 3).ToList(); //selection from saved data
    }

    private void AddOrRemoveMyList(List<SomeType> someList)
    {
        myList = someList;
    }

    private async Task ResetSelection()
    {
        myList = new List<SomeType>(); //<- dont even needed as the preselection in OpenModal() works allready like intended on opening

        await myModal.Hide();
    }
```
**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Blazorize 0.9.5.2
net core sdk 5.0.404
net core runtime 5.0.13

I Just tested it with my "Datagrid Clear" example and can reproduce the error with it (example and reproduction steps added to the original post).
Note that in my real life project i make "heavy" use of Virtualized Datagrids so that the error pretty much blowed up my log even while testing it just with another person. (thats why i even stumbled on it)

So sometimes "one" error looks like this wall of error log.

> 28.12.2021 14:47:49 [WARN] [Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer] - Unhandled exception rendering component: A task was canceled. 
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> 28.12.2021 14:47:49 [WARN] [Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer] - Unhandled exception rendering component: A task was canceled. 
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> 28.12.2021 14:47:49 [WARN] [Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer] - Unhandled exception rendering component: A task was canceled. 
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> 28.12.2021 14:47:49 [WARN] [Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer] - Unhandled exception rendering component: A task was canceled. 
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> 28.12.2021 14:47:49 [WARN] [Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer] - Unhandled exception rendering component: A task was canceled. 
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> 28.12.2021 14:47:49 [ERROR] [Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost] - Unhandled exception in circuit 'oJ-H4WIJdzJT0GknmJIcU2RyC39qdFjLCxQwvUuH5Yg'. 
> System.AggregateException: Exceptions were encountered while disposing components. (A task was canceled.) (A task was canceled.) (A task was canceled.) (A task was canceled.) (A task was canceled.)
>  ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
>    --- End of inner exception stack trace ---
>  ---> (Inner Exception #1) System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)<---
> 
>  ---> (Inner Exception #2) System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)<---
> 
>  ---> (Inner Exception #3) System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)<---
> 
>  ---> (Inner Exception #4) System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)<---
> 
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
> System.Threading.Tasks.TaskCanceledException: A task was canceled.
>    at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
>    at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
>    at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
>    at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
>

--- a/Documentation/Blazorise.Docs/Pages/Docs/Extensions/DataGrid/BindingDataPage.razor
+++ b/Documentation/Blazorise.Docs/Pages/Docs/Extensions/DataGrid/BindingDataPage.razor
@@ -53,6 +53,13 @@
                 Note that this mode requires that all rows have the same height.
             </AlertDescription>
         </Alert>
+
+        <Alert Margin="Margin.Is4.FromTop" Color="Color.Info" Visible>
+            <AlertDescription>
+                Note that there is a known bug in .NET5 where Virtualize throws a <Code>TaskCanceledException</Code> on the background, this does not seem to crash your application.
+                <Blazorise.Link To="https://github.com/dotnet/aspnetcore/issues/28280">TaskCanceledException when scrolling Virtualized component #28280</Blazorise.Link>
+            </AlertDescription>
+        </Alert>
     </DocsPageSectionHeader>
     <DocsPageSectionContent>
         <DataGridVirtualizeExample />

